### ISSUE TITLE
InvalidJavadocPosition: false positive when comment is between javadoc and package

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#7430]
````
$ cat TestClass.java
/** 1. **/
/** 2. **/
// abc
package a;


$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="InvalidJavadocPositionCheck" />
    </module>
</module>

$ java -jar checkstyle-8.28-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1:1: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
[ERROR] TestClass.java:2:1: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
Audit done.
Checkstyle ends with 2 errors.
````

Violation on line 1 is valid. There should be no error on line 2.

Javadoc tool seems to always choose the javadoc closest to the target. This is why 2 is valid and 1 isn't.

**Comment by rnveach** [comment]
This is different than https://github.com/checkstyle/checkstyle/issues/43 because of the code being used in checkstyle. In the other issue we have a token and are searching for the javadoc. This code we have a javadoc and are looking for the target to see if it is in the correct position.

**Comment by romani** [comment]
@rnveach , why you think we should report violation on 1st comment. 
There are cases then user use licence header in javadoc format, and here is just another example - http://rveach.no-ip.org/checkstyle/regression/reports/285/pmd/index.html#A1
I remember there were other cases when user complained about this.

It is kind hard decision when we should do same as javadoc tool and when behave differently.

**Comment by rnveach** [comment]
@romani As stated above, first comment is not a javadoc and will not be sent to the html report in the javadoc tool. https://github.com/checkstyle/checkstyle/issues/7430#issuecomment-569971860 I confirmed this locally before issue. If users want this in the javadoc they should merge the 2.

**Comment by romani** [comment]
problem is when user do not want to merge comments: first comment is license header, second is javadoc.
as we have in PMD case
```
/**
   * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
  */
/**
   * Here is some documentation details .....  
  */
// stolen from XPath Explorer (http://www.xpathexplorer.com)
package net.sourceforge.pmd.cpd;
```

**Comment by rnveach** [comment]
It doesn't change that it isn't a valid javadoc. It will not go to html report regardless of what user wants. Purpose of making it a javadoc is so it is treated as one but no tool will do that. Tool will only pick the javadoc closest to the target. All others are ignored like regular comments.

**Comment by romani** [comment]
I also can not find .... ok lets make it our Check stricter than tool.

**Comment by rnveach** [comment]
> ok lets make it our Check stricter than tool.

Its not stricter. Tool will not see 1 as a javadoc and will only use 2.

**Comment by romani** [comment]
fix is merged.
